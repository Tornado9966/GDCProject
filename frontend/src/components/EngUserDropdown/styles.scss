@import 'assets/_colors.scss';

.ui {
  &.user-dropdown {
    padding: 5px;
    &:hover {
      background: rgba(48, 48, 48, 0.1);
      padding: 5px;
      border-radius: 5px;
    }
  }
}