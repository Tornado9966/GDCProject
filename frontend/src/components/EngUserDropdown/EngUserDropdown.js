import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Dropdown, Image } from 'semantic-ui-react';

import * as constText from 'constants/text-constants';
import defaultIcon from './defaultIcon.png';

import './styles.scss';

export function EngUserDropdown({userImage, firstName, logout}) {
  const trigger = (
      <span>
          <Image avatar src={userImage || defaultIcon} />
      </span>
  );
  
  const options = [
    { 
      key: 'user',
      text: (
        <span>
          {constText.signedInAs} 
          <strong>{firstName}</strong>
        </span>
      ),
      disabled: true,
    },
    { key: constText.profile, text: constText.profile, as: Link, to: constText.linkToProfile },
    { key: constText.orders, text: constText.orders, as: Link, to: constText.linkToOrders },
    { key: constText.signOutKey, text: constText.signOut, onClick: logout }
  ];
  
  return (
      <Dropdown 
        trigger={trigger} 
        direction='left' 
        options={options}
        className='user-dropdown'
      />
  );
}

EngUserDropdown.propTypes = {
    userImage: PropTypes.string,
    userName: PropTypes.string,
    logout: PropTypes.func
};

EngUserDropdown.defaultProps = {
    userImage: defaultIcon,
    userName: 'anonymous',
    logout: ()=>{}
};