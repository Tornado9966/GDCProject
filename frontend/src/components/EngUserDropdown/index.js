import {connect} from 'react-redux';

import { userActions } from 'redux/actions';
import { EngUserDropdown } from './EngUserDropdown';

function mapState(state={}) {
  const { firstName, userImage } = state.auth;
  return {
    firstName,
    userImage
  };
}

const mapDispatchToProps = {
  logout: userActions.logout
};
  
export default connect(mapState, mapDispatchToProps)(EngUserDropdown);
