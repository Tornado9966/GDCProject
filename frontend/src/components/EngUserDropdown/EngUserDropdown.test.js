import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';

import { EngUserDropdown} from './EngUserDropdown';

const setUp = (props = {}) => {
    const component = shallow(<EngUserDropdown {...props} />);
    return component;
};

describe('EngUserDropdown', () => {
    let component;
    const loggedIn = true;

    beforeEach(() => {
        component = setUp();
    });

    it('matches the snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<EngUserDropdown debug />);
        expect(component).toMatchSnapshot();
    });
    
    it('should render correctly with no props', () => {
        const component = shallow(<EngUserDropdown />);
        
        expect(component).toMatchSnapshot();
    });

    it('should render correctly with boolean prop', () => {
        const component = shallow(<EngUserDropdown loggedIn={loggedIn} />);
        expect(component).toMatchSnapshot();
    });

    it('it should contain user-dropdown element', () => {
        const wrapper = component.find('.user-dropdown');
        expect(wrapper.length).toBe(1);
    });

    it('it should contain dropdown', () => {
        const wrapper = mount(
          <MemoryRouter>
            <EngUserDropdown loggedIn={loggedIn} />
          </MemoryRouter>);
        expect(wrapper.find('div[role="listbox"]').length).toBe(1);
    });
});