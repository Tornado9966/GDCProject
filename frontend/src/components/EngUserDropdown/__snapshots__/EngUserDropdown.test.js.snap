// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngUserDropdown matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`EngUserDropdown renders correctly 1`] = `ShallowWrapper {}`;

exports[`EngUserDropdown should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`EngUserDropdown should render correctly with boolean prop 1`] = `ShallowWrapper {}`;

exports[`EngUserDropdown should render correctly with no props 1`] = `ShallowWrapper {}`;
